src/Symfony/Component/Form/Extension/Core/Type/FormType.php

```diff
@@ -79,13 +78,13 @@ public function buildView(FormView $view, FormInterface $form, array $options)
             }
 
             // Complex fields are read-only if they themselves or their parents are.
-            if (!$readOnly) {
-                $readOnly = $view->parent->vars['read_only'];
+            if (!isset($view->vars['attr']['readonly']) && isset($view->parent->vars['attr']['readonly']) && false !== $view->parent->vars['attr']['readonly']) {
+                $view->vars['attr']['readonly'] = true;
             }
```

Comment:
@mnapoli Ok so to avoid the BC break, we just need to add a line here with:
```php
$view->vars['read_only'] = true; // BC`
```
Then revert it in 3.0.

Would you like to send a PR ?